transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionNotifierTestCase.java

```diff
@@ -0,0 +1,218 @@
+
+    private TestReconnectNotifier mockNotifier = null;
+
+
+    protected String getConfigFile()
+    {
+        return "ftp-reconnection-notifier-fails-config.xml";
+    }
+
+    @Test
+    public void testNotificationOnReconnectionAttemptFailure() throws Exception
+    {
+        mockNotifier = mock(TestReconnectNotifier.class);
+        Connector c = muleContext.getRegistry().lookupConnector("FTP");
+        assertNotNull(c);
+        RetryPolicyTemplate rpf = c.getRetryPolicyTemplate();
+        assertNotNull(rpf);
+        rpf.setNotifier(mockNotifier);
+        MuleFtpServer server = new MuleFtpServer(2221, 1);
+        server.start();
+        testServerConnected();
+        server.stop();
+        testServerDisconnected();
+    }
+
+    private void verifyOnSuccessNotification()
+    {
+        PollingProber pollingProber = new PollingProber(POLLING_FREQUENCY * 5, POLLING_FREQUENCY);
+        pollingProber.check(new JUnitProbe()
+        {
+            @Override
+            protected boolean test() throws Exception
+            {
+                verify(mockNotifier, times(1)).onSuccess(any(RetryContext.class));
+                return true;
+            }
+
+            @Override
+            public String describeFailure()
+            {
+                return "Success notification wasn't triggered";
+            }
+        });
+    }
+
+    private void verifyOnFailureNotification()
+    {
+        PollingProber pollingProber = new PollingProber(POLLING_FREQUENCY * 5, POLLING_FREQUENCY);
+        pollingProber.check(new JUnitProbe()
+        {
+            @Override
+            protected boolean test() throws Exception
+            {
+                verify(mockNotifier, atLeast(1)).onFailure(any(RetryContext.class), any(Throwable.class));
+                return true;
+            }
+
+            @Override
+            public String describeFailure()
+            {
+                return "Failure notification wasn't triggered";
+            }
+        });
+    }
+
+    private void testServerConnected() throws Exception
+    {
+        runFlow("main-test");
+        verifyOnSuccessNotification();
+    }
+
+    private void testServerDisconnected()
+    {
+        try
+        {
+            runFlow("main-test");
+        }
+        catch (Exception e)
+        {
+            verifyOnFailureNotification();
+        }
+    }
+
+
+    public static class MuleFtpServer
+    {
+        private FtpServer server;
+        FtpServerFactory serverFactory;
+        ListenerFactory factory;
+        public MuleFtpServer(int port, long delay)
+        {
+
+            serverFactory = new FtpServerFactory();
+            factory = new ListenerFactory();
+            factory.setPort(port);
+            serverFactory.addListener("default", factory.createListener());
+            serverFactory.addListener("default", factory.createListener());
+            PropertiesUserManagerFactory userManagerFactory = new PropertiesUserManagerFactory();
```

Comment:
Is all this configuration required? I don't see usages of setHomeDirectory, for example, in existing tests. Also existing test do not add a listener, they use a default one